iob_soc.py

```diff
@@ -132,15 +137,13 @@ def _create_submodules_list(cls, extra_submodules=[]):
             # Software modules
             printf,
         ]
-        if "USE_EXTMEM" in sys.argv:
-            submodule_list += [
-                iob_cache,
-                # Modules required for CACHE
-                (iob_ram_2p, {"purpose": "simulation"}),
-                (iob_ram_2p, {"purpose": "fpga"}),
-                (iob_ram_sp, {"purpose": "simulation"}),
-                (iob_ram_sp, {"purpose": "fpga"}),
-            ]
+        if "INIT_MEM" in sys.argv:
+            # Remove the submodule entries
+            submodule_list.remove(iob_cache)
+            submodule_list.remove((iob_ram_2p, {"purpose": "simulation"}))
+            submodule_list.remove((iob_ram_2p, {"purpose": "fpga"}))
+            submodule_list.remove((iob_ram_sp, {"purpose": "simulation"}))
+            submodule_list.remove((iob_ram_sp, {"purpose": "fpga"}))
```

Comment:
adding or not cache submodules should depend on `USE_EXTMEM` configuration.

Reply:
yes i didint uderstand the cache error and though that maybe the error was related to this, thats why i changed, but its already reverted